src/hal/hal_i2c_microSpecific.c

```diff
@@ -0,0 +1,77 @@
+#include "hal.h"
+#include "hal_i2c.h"
+
+#include "hal_i2c_microSpecific.h"
+#include "stm32l031xx.h"
+#include "stm32l0xx.h"
+#include "system_stm32l0xx.h"
+
+#include "shared_defs.h"
+
+
+hal_error_E hal_i2c_microSpecific_init(void) {
+
+    //This is to configure it in master mode
+
+    //clear PE bit
+    I2C1->CR1 &= ~(1<<I2C_CR1_PE_Pos);
+
+    //Enable analog filter
+    I2C1->CR1 &= ~(1<<I2C_CR1_ANFOFF_Pos);
+
+    //Disable digital filter
+    I2C1->CR1 &= ~(I2C_CR1_DNF_Msk);
+
+    //Configure timings
+    //Generated with CubeMX might not be best practice
+    I2C1->TIMINGR = 0x00B07CB4;
+
+    //NOSTRETCH must be kept clear in master mode
+    I2C1->CR1 &= ~(1<<I2C_CR1_NOSTRETCH_Pos);
+
+    //Enable the peripheral
+    I2C1->CR1 |= 1 << I2C_CR1_PE_Pos;
+}
+
+hal_error_E hal_i2c_sendData(uint8_t* data, uint8_t nBytes) {
```

Comment:
I think we will want to refactor this to be channelized like hal_uart. So these microspecific functions would be made private and passed as pointer to the shared module through a config struct. Then hal_i2c_sendData (and the similar functions defined below) would live in the hal repo, with just the register diddling living here.

This also provides flexibility for the multi i2C bus use case.

Reply:
We probably also want to think through the master/slave cases and make sure these functions return the correct error codes.